core-bundle/src/Resources/contao/library/Contao/Database/Statement.php

```diff
@@ -308,6 +309,15 @@ static function ($varParam)
 		return new Result($this->statement, $this->strQuery);
 	}
 
+	/**
+	 * @return int
+	 */
+	private function countQueryTokens()
+	{
+		// Backwards compatibility
```

Comment:
How is this a BC layer?

Reply:
It counts the tokens the way the old implementation did which is error prone and only needed for deprecated calls with too many arguments.

Reply:
How will we count the tokens in the new implementation?

Reply:
We will not. Once the deprecation layer is removed, for a mismatch between the number of tokens and arguments the database will throw an exception.

Reply:
What do you think about https://github.com/ausi/contao-contao/pull/1? I find it more clear this way.

Reply:
But I should probably add a comment that clarifies this.

Reply:
Added a comment in e7d12adb8ebdabbc55563001c8e5278bb63ad98f

Reply:
WDYT about https://github.com/ausi/contao-contao/pull/2?